### PR TITLE
go-tools: add go-deadlock dep

### PR DESCRIPTION
**Pull request by HunDunDM**
Signed-off-by: Zheng Xiangsheng <hundundm@gmail.com>

### What problem does this PR solve? 

We used `go-deadlock` in the test but did not add it as a dependency on `go.mod`.

### What is changed and how it works?

Add it as a dependency on `go.mod`.

### Check List 

Tests 
 - Manual test (add detailed scripts or steps below)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -5,6 +5,7 @@ cache:
     - $HOME/.cache/go-build
     - $GOPATH/pkg/mod
     - $GOPATH/src/github.com/pingcap/pd/.retools
```
Comment (rleungx) [inline .travis.yml:7]: Do we still need it?

Comment (HunDunDM) [reply .travis.yml:7]: yes

Comment (lhy1024) [reply .travis.yml:7]: it will be better to add comment here

**Pull request merged**